### PR TITLE
Add(#5): Add oracle db support

### PR DESCRIPTION
**Pull request by indiVar0508**
Add oracle DB in github action job

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -88,8 +92,55 @@ jobs:
...
+  test-oracle:
+    name: test-python${{ matrix.python-version }}-oracle
+    strategy:
+      matrix:
+        python-version:
+         - "3.6"
+         - "3.10"
+    services:
+      oracle:
+        image: gvenzl/oracle-xe
```
Comment (AbdealiLoKo) [inline .github/workflows/main.yml:112]: Which version of oracle does this use ?
Can we use the oldest version available ?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -88,8 +92,55 @@ jobs:
...
+        ports:
+          - 1521:1521
+        env:
+          ORACLE_PASSWORD: p#ssw0rd
```
Comment (AbdealiLoKo) [inline .github/workflows/main.yml:116]: Can we keep same password for all DBs ?

File: benchmark.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class Article(Model):
         __tablename__ = "article"
         __versioned__ = copy(options)
 
-        id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+        id = sa.Column(sa.Integer, sa.schema.Sequence("article_id_seq"), autoincrement=True, primary_key=True)
```
Comment (AbdealiLoKo) [inline benchmark.py:56]: Can we use identity instead ?

File: .github/workflows/main.yml (hunk 1 above)
Comment (indiVar0508) [reply .github/workflows/main.yml]: It was currently 21, downgraded to 18 for now.

File: benchmark.py (hunk 3 above)
Comment (indiVar0508) [reply benchmark.py]: Updated to Identity for all testcases there were certain cases where Identity wasn't working have seperated them out in commits and explained the issue faced.

Comment (indiVar0508) [reply benchmark.py]: It seems Identity column is causing issues with MSSQL DB where it requires `IDENTITY_INSERT` to be ON

**Comment by indiVar0508** [comment]
Closing this PR as Oracle Action job is added as part of #47 and a separate PR will be raised to address failing test case issue for Oracle DB as it has impact on MSSQL DB when tested with Identity object and subquery issue

**Pull request closed without merging**